lib/net/validations.rb

```diff
@@ -119,6 +119,12 @@ def self.normalize_ip6(ip)
       IPAddr.new(ip, Socket::AF_INET6).to_s rescue ip
     end
 
+    def self.normalize_network_address(ip, mask)
+      IPAddr.new(ip).mask(mask).to_s
```

Comment:
```suggestion
      IPAddr.new("#{ip}/#{mask}").to_s
```
Saves creating one IPAddr instance by directly using the right mask.

Reply:
`to_s` compresses v6 addresses:

```console
irb(main):007:0> IPAddr.new('2600::0').to_s
=> "2600::"
```

not sure this is intended here.

Reply:
I do think it's good to compress. Makes it easier to read.

Reply:
While I technically agree (and would always use the compressed notation), I also kinda think we shouldn't override the users preference?

Reply:
That did make me think about where to solve it. In https://github.com/theforeman/foreman/pull/8528#issuecomment-842221068 I made the argument that we can also normalize it when using the value.

On the other hand, now you can have multiple Subnet instances represent the same subnet (`2600::0` and `2600::`) without the DB knowing. Now that we're PostgreSQL only, we can investigate moving to the native IP representation so that they're also properly sorted etc, but that's probably a bigger task. I think that the DB model layer also returns an IPAddr instance then which the Foreman code may not handle right now.

Reply:
I am not replacing two fast method calls with an interpolation. That is actually slower and more error prone :-)

Anyway, valid point to the compression. While I lean towards doing it, in case you don't like it then I can turn this into validation so an error is thrown if there are any bits in the non-mask section of the address.